modules/402-ingress-nginx/hooks/alert_deprecated_geoip_version.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook/metrics"
+	"github.com/flant/addon-operator/sdk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+
+	"github.com/deckhouse/deckhouse/go_lib/dependency"
+	"github.com/deckhouse/deckhouse/go_lib/dependency/k8s"
+)
+
+// TODO: Remove this migration hook after release 1.65
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/ingress-nginx/deprecated_geoip_version",
+	Schedule: []go_hook.ScheduleConfig{
+		{Name: "cron", Crontab: "0 * * * *"},
+	},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:                         "controller",
+			ApiVersion:                   "deckhouse.io/v1",
+			Kind:                         "IngressNginxController",
+			ExecuteHookOnSynchronization: pointer.Bool(true),
+			ExecuteHookOnEvents:          pointer.Bool(false),
+			FilterFunc:                   inletHostWithFailoverFilter,
+		},
+	},
+}, dependency.WithExternalDependencies(searchForDeprecatedGeoip))
+
+var (
+	metricsGroup            = "d8_deprecated_geoip_version"
+	ingressNamespace        = "d8-ingress-nginx"
+	ingressAnnotationPrefix = "nginx.ingress.kubernetes.io/"
+
+	eolVersion = semver.MustParse("1.10")
+
+	geoipVarsRegexp = regexp.MustCompile(`\$geoip_(country_(code3|code|name)|area_code|city_continent_code|city_country_(code3|code|name)|dma_code|latitude|longitude|region|region_name|city|postal_code|org)([^_a-zA-Z0-9]|$)+`)
+
+	// objectBatchSize - how many ingress objects to list from k8s at once
+	objectBatchSize = int64(30)
+	// fetchSecretsInterval pause between fetching ingress objects from apiserver
+	fetchSecretsInterval = 1 * time.Second
+)
+
+type ControllerVersion struct {
```

Comment:
```suggestion
type controllerVersion struct {
```

don't think we have to export this